### PR TITLE
SITES-993: Stop the infinite redirect on 403 pages

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Fixes login issues with lbredocs.stanford.edu

# Needed By (Date)
- ?

# Urgency
- ?

# Steps to Test

1. Add this work to a branch on ACSF-CardinalD7
2. Clone lbredocs.stanford.edu production site down to dev or test
3. Log in as user 1
4. Add `rajesh12` to the allowed sunet users
5. Log out and log back in as jbickar and get redirect loop
6. Log back in as user 1
7. add `itservices:webservices` to allowed workgroups
8. Log out and log back in as jbickar and get logged in
9. Remove itservices:webservices from allowed workgroups
10. Deploy new code 
11. Log in as jbickar 

# Affected Projects or Products
- Sites wholly behind authentication

# Associated Issues and/or People
- SITES-993
- LBRE
- INC00882041

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jbickar** [comment]
Hm, I'm getting "502 Bad Gateway" when logging in to https://lbredocs-test.sites.stanford.edu/

File: stanford_ssp.module [hunk 1]
```diff
@@ -283,9 +282,9 @@ function stanford_ssp_page_build(array &$page) {
   // We don't want to mess with the 403 page setting so we should check the
   // header response on the page build.
   $headers = drupal_get_http_header();
-  if (user_is_anonymous() && isset($headers["status"]) && $headers["status"] == "403 Forbidden") {
-    if (variable_get("stanford_ssp_automagic_login", FALSE)) {
 
+  if (user_is_anonymous() && isset($headers["status"]) && $headers["status"] == "403 Forbidden" && !preg_match('/login\.stanford\.edu/', $_SERVER['HTTP_REFERER'])) {
```
Comment (jbickar) [inline stanford_ssp.module]: what about `login-uat.stanford.edu`?

Comment (jbickar) [reply stanford_ssp.module]: I changed this to:
```
  if (user_is_anonymous() && isset($headers["status"]) && $headers["status"] == "403 Forbidden" && !preg_match('/login(-uat)?\.stanford\.edu/', $_SERVER['HTTP_REFERER'])) {
```
Yet I still get "502 Bad Gateway" with only a workgroup that I'm in allowed. If I add myself to "Allowed SUNetIDs", I can log in.

2019-04-02-151655-screenshot

**Comment by jbickar** [comment]
@sherakama, this works on ACP, but I'm still getting `502 Bad Gateway` on https://lbredocs-test.sites.stanford.edu/sso/login with the following configuration:

2019-06-14-72123-screenshot

**Comment by sherakama** [comment]
I am seeing an additional issue on that site where SSP is unable to communicate with the workgroup API.

**Comment by sherakama** [comment]
@jbickar now that @imonroe is all set up locally this might be a good one for him to pick up on.

**Comment by jbickar** [comment]
Yassssss!

**Comment by sherakama** [comment]
@jbickar since we are going to be talking about SimpleSAML in dev chat, perhaps, @imonroe can circle back on this task after.

**Comment by sherakama** [comment]
Moving to https://github.com/SU-SWS/stanford_ssp/pull/95

**Pull request closed without merging**